### PR TITLE
Remove Truffle support from TS client

### PR DESCRIPTION
**Pull request by aefhm**
## Description
Consensys is [sunsetting](https://consensys.io/blog/consensys-announces-the-sunset-of-truffle-and-ganache-and-new-hardhat) Truffle.

Unblock https://github.com/oasisprotocol/sapphire-paratime/pull/173 and dependent on  https://github.com/oasisprotocol/docs/pull/547

### CODE REVIEW

**Review by matevz — CHANGES_REQUESTED** [review]
Also delete examples/truffle.

File: examples/truffle/package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "@truffle/hdwallet-provider": "^2.0.14"
   },
   "dependencies": {
-    "@oasisprotocol/sapphire-paratime": "workspace:^",
+    "@oasisprotocol/sapphire-paratime": "~1.1.1",
```
Comment (nhynes) [inline examples/truffle/package.json:16]: we're pinning support for truffle to this version now? why not keep it 1.x, as we'll need to cut a 2.x at some point

**Review by nhynes — APPROVED** [review]

File: examples/truffle/package.json (hunk 1 above)
Comment (aefhm) [reply examples/truffle/package.json]: We plan on a patch release `1.1.2` with #175 and various other QOL issues.

This PR #191 will land in some `2.0.0` or `1.2.0` depending on how we like to name things.

**Comment by matevz** [comment]
https://github.com/oasisprotocol/docs/pull/547 should be merged before truffle removal.

**Comment by aefhm** [comment]
I think this PR is ready for next major release. 🙏 

cc @matevz

File: clients/js/src/compat.ts [hunk 2]
```diff
@@ -653,42 +640,7 @@ export async function fetchRuntimePublicKey(
         ) => Promise<any> | void;
       };
 
-      // For Truffle, turn a callback into an synchronous call
```
Comment (matevz) [inline clients/js/src/compat.ts:656]: Did you test the removal of this with Brave and Metamask? EIP-1193 is "Ethereum Provider JavaScript API", so this is not just truffle.

**Review by matevz — CHANGES_REQUESTED** [review]

File: clients/js/src/compat.ts (hunk 2 above)
Comment (aefhm) [reply clients/js/src/compat.ts:656]: Test flow of using Metamask on Brave with our example through the Hardhat [boilerplate](https://docs.oasis.io/dapp/sapphire/browser/) right?

That flow does work, but now I'm suspicious of my local dev and testing. Good callout. Lemme keep digging. 🤔

Comment (aefhm) [reply clients/js/src/compat.ts:656]: K. I see the error now. We had another fallback mechanism, but I think the client should be EIP-1193 compliant as you corrected! 👍

**Review by matevz — APPROVED** [review]
I tested demo-voting with MetaMask using the new client and everything seems to work. Good job.

**Pull request merged**